Add 2.5 templates including Push Model agent changes

sarroutbi opened the pull request:
# PR Title 
Add 2.5 templates to include Push Model agent changes

## Type of Change
*(Select all that apply)*
- [ ] Bug fix (non-breaking change)
- [x] New feature (non-breaking change)
- [ ] Breaking change (fix/feature causing existing behavior to change)
- [ ] Documentation update (standalone)
- [ ] Code refactor (no functional changes)
- [ ] Test cases (added/modified)
- [ ] CI/CD changes
- [ ] Other (please specify: ______)

## Related Issues

## Change Description

### Concise Summary

This commit introduces the necessary configuration template files for version 2.5. These new templates add the required configuration options to support the new agent operating in push mode.

The update follows the established configuration upgrade process by creating a new version directory and defining a non-breaking, minor version update.

### Technical Details
 
Configuration changes motivated from Push Model implementation:
- Includes configuration to access Verifier.
- Includes configuration related to exponential back-off.

## Documentation Updates Required
*(Check all that apply)*
- [ ] Updated markdown docs (file path: ______)
- [ ] Updated code comments/docstrings
- [ ] Needs user guide updates (`docs/`)
- [ ] Needs ReadTheDocs updates
- [ ] No docs needed (requires maintainer approval)

## Verification Process
New configuration values can be read from agent.

## Checklist
- [x] Code follows project style guidelines
- [x] Unit/integration tests added/updated
- [ ] Documentation updated (per above section)
- [x] Commit messages follow [Chris Beams' How to Write a Git Commit Message article] (https://chris.beams.io/posts/git-commit/)
- [ ] CHANGELOG updated (if applicable)
- [x] All tests pass (local & CI)

## Additional Context
*(Optional: follow-up tasks, special considerations)*

sarroutbi commented:
> The following options are not present in the agent configuration:
> 
>     * `verifier_ip`
> 
>     * `verifier_port`

Should not we add them? I understand this information should be available in configuration, and way to connect to Verifier should not be exclusive to command line arguments. Apart from that ... it might be better to use something like `verifier_base_url` to allow setting `http://` or `https://`. WDYT?
> 
> 
> There are many missing options added for the push-attestation agent:
> 
>     * `certification_keys_server_identifier`
> 
>     * `disabled_signing_algorithms`
> 
>     * `ima_logs_appendable`
> 
>     * `ima_logs_formats`
> 
>     * `ima_logs_supports_partial_access`
> 
>     * `ima_ml_count_file`
> 
>     * `registrar_api_versions`
> 
>     * `uefi_logs_appendable`
> 
>     * `uefi_logs_evidence_version`
> 
>     * `uefi_logs_formats`
> 
>     * `uefi_logs_supports_partial_access`
> 
>     * `uefi_logs_binary_file_path`
> 

Sure. I'll add them

> 
> There is also an old option that is missing from all versions and we could add:
> 
>     * `agent_data_path`

I'll add it too

> 
> 
> For the verifier, there are also these that I'm aware of, but we could add later as part of the push-attestation branch:
> 
>     * `mode`
> 
>     * `challenge_lifetime`

ansasaki commented:
> Should not we add them? I understand this information should be available in configuration, and way to connect to Verifier should not be exclusive to command line arguments. Apart from that ... it might be better to use something like verifier_base_url to allow setting http:// or https://. WDYT?

I'm OK adding these verifier-related options to the templates as long as they are also added to the agent code to be actually used.

sarroutbi commented:
> > Should not we add them? I understand this information should be available in configuration, and way to connect to Verifier should not be exclusive to command line arguments. Apart from that ... it might be better to use something like verifier_base_url to allow setting http:// or https://. WDYT?
> 
> I'm OK adding these verifier-related options to the templates as long as they are also added to the agent code to be actually used.

I added `verifier_url` to the Agent configuration for this purpose

sarroutbi commented:
/packit retest-failed

sarroutbi commented:
/packit retest-failed

stringlytyped commented:
For the agent template:

- If we are adding a `verifier_url` option (which I agree we should), we should have a `registrar_url` option to match.
- If either `verifier_url` or `registrar_url` has a scheme of `http://`, that should be logged as a warning on startup.

For the verifier template:

- `mode` and `verification_timeout` look good
- `nonce_lifetime` should be `challenge_lifetime`
- We should add a `verifier_url` option to mirror the agent configuration so that the verifier can report URLs with the correct hostname/IP rather than bind address

For the registrar template:

- I am not seeing any changes from the previous version. I'm not missing anything, right?
- We should add `mode` here too in order to disable certain features when push is used (e.g., registration over insecure HTTP, payload feature, etc.)
- Again, I think we should add a `registrar_url` option for the same reason as above

@ansasaki A couple questions for you:

- Would it be feasible to rename `quote_interval` to `attestation_interval` so that it is more generic and makes sense for non-TPM uses?
- Once this is merged, can I make edits to the template to add more detailed documentation such as which options are push/pull specific?

ansasaki commented on templates/2.5/verifier.j2:
```diff
@@ -0,0 +1,261 @@
+mode = {{ verifier_mode }}
+#
+# Lifetime in seconds for nonces sent to agents in push mode.
+nonce_lifetime = {{ verifier_nonce_lifetime }}
```
This should be `challenge_lifetime`

ansasaki commented on templates/2.5/mapping.json:
```diff
@@ -0,0 +1,33 @@
+        "verifier": {
+            "add": {
+                "mode": "pull",
+                "nonce_lifetime": "1800",
```
This should be `challenge_lifetime`

ansasaki reviewed (changes requested):
Sorry @sarroutbi, could you please replace `nonce_lifetime` with `challenge_lifetime` and push again?
